src/library/scala/collection/TraversableLike.scala

```diff
@@ -142,7 +142,7 @@ trait TraversableLike[+A, +Repr] extends Any
 
   def ++[B >: A, That](that: GenTraversableOnce[B])(implicit bf: CanBuildFrom[Repr, B, That]): That = {
     val b = bf(repr)
-    if (that.isInstanceOf[IndexedSeqLike[_, _]]) b.sizeHint(this, that.seq.size)
+    if (that.isInstanceOf[IndexedSeqLike[_, _]]) sizeHintTo(b, that.seq.size)
```

Comment:
Don't you need to remove this if and replace it with "sizeHintTo" (that will do nothing in case that 'that' isn't an IndexedSeqLike)?